src/Parser/Mage/Fire/Modules/Items/MarqueeBindingsOfTheSunKing.js

```diff
@@ -0,0 +1,151 @@
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+      return;
+    }
+    this.hasBindingsBuff = true;
+    this.buffUsed = false;
+    this.wastedProcs += 1;
+    this.totalProcs += 1;
+    debug && console.log("Buff Refreshed " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+  }
+
+  on_byPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+      return;
+    }
+    this.hasBindingsBuff = false;
+    debug && console.log("Buff Removed @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    if (this.buffUsed === false) {
+      this.wastedProcs += 1;
+      debug && console.log("Buff Expired @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    } else {
+      this.usedProcs += 1;
+    }
+  }
+
+  on_byPlayer_begincast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id) {
+      return;
+    }
+    this.beginCastTimestamp = event.timestamp;
+    this.beginCastFound = true;
+
+    debug && console.log("Pyroblast Begin Cast @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id || this.beginCastFound === false) {
+      return;
+    }
+    this.beginCastFound = false;
+    debug && console.log("Pyroblast Casted @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    this.castTimestamp = event.timestamp;
+    const castTime = this.castTimestamp - this.beginCastTimestamp;
+    if (castTime >= CAST_BUFFER && this.hasBindingsBuff === true) {
+      this.isBuffed = true;
+      this.buffUsed = true;
+      debug && console.log("Buff Used @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id) {
+      return;
+    }
+      if (this.isBuffed === true) {
+        this.damage += getDamageBonus(event, BONUS_DAMAGE);
+        this.isBuffed = false;
+        debug && console.log("Pyroblast Damage @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      } else {
+        debug && console.log("Non Buffed Pyroblast Damage @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      }
+  }
+
+  get avgBonusDamage() {
+    return this.damage / this.usedProcs;
+  }
+
+  on_finished() {
+    if (this.combatants.selected.hasBuff(SPELLS.KAELTHAS_ULTIMATE_ABILITY.id)) {
+      const adjustedFightEnding = this.owner.currentTimestamp - 5000;
+      if (this.buffAppliedTimestamp < adjustedFightEnding) {
+        this.wastedProcs += 1;
+        debug && console.log("Fight Ended with Unused Proc @ " + formatMilliseconds(this.owner.currentTimestamp - this.owner.fight.start_time));
```

Comment:
This seems a little harsh, and possibly confusing that this counts as wasted... Still I guess it's arguably the right thing to do.

Reply:
Yea i gave 5 seconds at the end of the fight so you could reasonably get the buff, realize you have it, and cast it before the fight ends ... but i might increase that.

Essentially that disregards any procs that were applied within the last x seconds of the fight.